### PR TITLE
Add `static_set::retrieve_all`

### PR DESCRIPTION
**Pull request by PointKernel**
This PR adds the `static_set::retrieve_all` function. It also changes the default CG size to 4 and the default window size to 1 for slightly better performance.

### CODE REVIEW

**Comment by PointKernel** [comment]
Notes: 
- using `__forceinline__` for functors like `slot_is_filled` and `get_slot` introduces about ±3% runtime performance difference at the maximum (see table below). The differences are most likely coming from runtime noises.
- Custom kernels with shared memory buffer cannot outperform the cub function.
- Performance of `static_set<int64_t>` is consistent with `static_map<int32_t, int32_t>`.

|  Key  |  Distribution  |  Occupancy  |   Ref Time |   Ref Noise |   Cmp Time |   Cmp Noise |        Diff |   %Diff |  Status  |
|-------|----------------|-------------|------------|-------------|------------|-------------|-------------|---------|----------|
|  I32  |     UNIQUE     |     0.1     |   9.976 ms |      14.04% |   9.654 ms |       6.95% | -322.012 us |  -3.23% |   PASS   |
|  I32  |     UNIQUE     |     0.2     |   5.187 ms |      23.91% |   5.214 ms |      20.49% |   26.805 us |   0.52% |   PASS   |
|  I32  |     UNIQUE     |     0.3     |   3.473 ms |       0.79% |   3.485 ms |       0.90% |   11.319 us |   0.33% |   PASS   |
|  I32  |     UNIQUE     |     0.4     |   2.744 ms |       0.70% |   2.753 ms |       0.70% |    8.308 us |   0.30% |   PASS   |
|  I32  |     UNIQUE     |     0.5     |   2.335 ms |       0.66% |   2.339 ms |       0.54% |    4.681 us |   0.20% |   PASS   |
|  I32  |     UNIQUE     |     0.6     |   2.097 ms |       0.51% |   2.175 ms |      49.71% |   77.499 us |   3.70% |   FAIL   |
|  I32  |     UNIQUE     |     0.7     |   1.949 ms |       0.42% |   1.956 ms |       1.42% |    6.796 us |   0.35% |   PASS   |
|  I32  |     UNIQUE     |     0.8     |   1.880 ms |      31.13% |   1.848 ms |       0.98% |  -31.429 us |  -1.67% |   FAIL   |
|  I32  |     UNIQUE     |     0.9     |   1.756 ms |       0.68% |   1.764 ms |       1.94% |    8.069 us |   0.46% |   PASS   |
|  I64  |     UNIQUE     |     0.1     |  17.829 ms |       0.25% |  17.943 ms |       0.28% |  113.455 us |   0.64% |   FAIL   |
|  I64  |     UNIQUE     |     0.2     |  10.593 ms |      25.00% |  10.285 ms |      14.22% | -307.692 us |  -2.90% |   PASS   |
|  I64  |     UNIQUE     |     0.3     |   6.994 ms |      11.25% |   7.020 ms |      11.48% |   25.966 us |   0.37% |   PASS   |
|  I64  |     UNIQUE     |     0.4     |   5.628 ms |       0.25% |   5.752 ms |      16.92% |  123.469 us |   2.19% |   FAIL   |
|  I64  |     UNIQUE     |     0.5     |   5.024 ms |      10.43% |   4.974 ms |       0.16% |  -49.234 us |  -0.98% |   FAIL   |
|  I64  |     UNIQUE     |     0.6     |   4.456 ms |       0.24% |   4.458 ms |       0.25% |    2.175 us |   0.05% |   PASS   |
|  I64  |     UNIQUE     |     0.7     |   4.174 ms |       0.24% |   4.183 ms |       0.81% |    8.940 us |   0.21% |   PASS   |
|  I64  |     UNIQUE     |     0.8     |   3.978 ms |       6.42% |   3.937 ms |       0.30% |  -41.623 us |  -1.05% |   FAIL   |
|  I64  |     UNIQUE     |     0.9     |   3.760 ms |       0.21% |   3.758 ms |       0.24% |   -1.967 us |  -0.05% |   PASS   |

File: benchmarks/hash_table/static_set/retrieve_all_bench.cu [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+#include <nvbench/nvbench.cuh>
+
+#include <thrust/device_vector.h>
+
+using namespace cuco::benchmark;
+using namespace cuco::utility;
+
+/**
+ * @brief A benchmark evaluating `cuco::static_set::contains` performance
```
Comment (sleeepyjack) [inline benchmarks/hash_table/static_set/retrieve_all_bench.cu]:
```suggestion
 * @brief A benchmark evaluating `cuco::static_set::retrieve_all` performance
```

File: include/cuco/detail/static_set/functors.cuh [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+  /**
+   * @brief Constructs `get_slot` functor with the given storage ref.
+   *
+   * @param s Input storage ref
+   */
+  get_slot(StorageRef s) : storage_{s} {}
+
+  /**
+   * @brief Indicates if the target slot `s` is filled.
```
Comment (sleeepyjack) [inline include/cuco/detail/static_set/functors.cuh]: This seems off

**Review by sleeepyjack — APPROVED** [review]

File: include/cuco/detail/static_set/functors.cuh (hunk 2 above)
Comment (PointKernel) [reply include/cuco/detail/static_set/functors.cuh]: Thanks. Fixed

**Pull request merged**